logcart_driver:
  hardware_interface:
    loop_hz: 50

joint_state_controller:
  type        : "joint_state_controller/JointStateController"
  publish_rate: 20

base_controller:
  type        : "diff_drive_controller/DiffDriveController"
  left_wheel  : 'left_wheel_joint'
  right_wheel : 'right_wheel_joint'
  publish_rate: 20
  #publish_cmd: true
  pose_covariance_diagonal : [0.00001, 0.00001, 1000000000000.0, 1000000000000.0, 1000000000000.0, 0.001]
  twist_covariance_diagonal: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0]

  wheel_separation : 0.22
  wheel_radius : 0.0325

  wheel_separation_multiplier: 1.0 # default: 1.0
  wheel_radius_multiplier    : 1.0 # default: 1.0

  cmd_vel_timeout: 1.0 # Period (in s) allowed between two successive velocity commands. After this delay, a zero speed command will be sent to the wheels

  base_frame_id: base_link

  linear:
    x:
      has_velocity_limits    : true
      max_velocity           : 1.0 # m/s
      min_velocity           : -1.0 # m/s
      has_acceleration_limits: true
      max_acceleration       : 1.0  # m/s^2
      min_acceleration       : -1.0 # m/s^2
      # has_jerk_limits        : true
      # max_jerk               : 0.1  # m/s^3
      # min_jerk               : -0.1 # m/s^3
      
  angular:
    z:
      has_velocity_limits    : false
      max_velocity           : 10.0  # rad/s
      has_acceleration_limits: true
      max_acceleration       : 2.0  # rad/s^2
      min_acceleration       : -2.0 # rad/s^2
